contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,562 @@
+
+ADMIN:
+
+```
+apiVersion: "v1"
+kind: "PersistentVolume"
+metadata:
+  name: gce-disk-1
+  annotations:
+    volume.beta.kubernetes.io/mount-options: "discard"
+Spec:
+  volumeType: block
+  capacity:
+    storage: "10Gi"
+  accessModes:
+    - "ReadWriteOnce"
+  gcePersistentDisk:
+    fsType: "block"
+    pdName: "gce-disk-1"
+```
+
+***If admin specifies volumeType: block + fstype: ext4 then they would have the default behavior of files on block ***
+***fsType values will be provisioner dependent. Block is suggested for development simplicity
+
+# Container Runtime considerations
+It is important the values that are passed to the container runtimes are valid and support the current implementation of these various runtimes. Listed below are a table of various runtime and the mapping of their values to what is passed from the mount. 
+
+| runtime engine    | runtime options  | accessMode       |  
+| --------------    |:----------------:| ----------------:|
+| docker/runc/rkt   |  mknod / RWM     | RWO              |
+| docker/runc/rkt   |       R          | ROX              |
+
+# Implementation Plan, Features & Milesones
+
+Phase 1: v1.8
+Feature: Pre-provisioned PVs to precreated devices 
+
+               Milestone 1: API changes
+
+               Milestone 2: Restricted Access 
+               
+               Milestone 3: Scrubbing device after PV is deleted
+               
+               Milestone 4: Changes to the mounter interface as today it is assumed 'file' as the default.
+               
+               Milestone 5: Expose volumeType to users via kubectl
+               
+               Milestone 6: Adds enable/disable configuration to securityContext in PSP (Pod Security Policy) similar to hostPath
+               
+               Milestone 7: Validate container runtime options with user specifcations as indicated in UC3
+	       
+	       Milestone 8: Container Runtime changes
+
+Phase 2:  v1.9
+Feature: Discovery of block devices 
+
+                Milestone 1: Dynamically provisioned PVs to dynamically allocated devices
+
+                Milestone 2: Privileged container concerns    
+
+Other considerations:
+
+               Reference volume driver change(Attach/Detach logic) for pre-provisioned PVs
+
+               Reference volume driver change(Provision/Delete logic) for dynamic provisioning
+
+## Mounter interface proposed design
+
+```
+ type BlockMounter interface {
+	Mounter
+ 	CanSupportFileOnBlock() error
+            // kubelet needs volume's device path when attaches a volume to container
+ 	GetVolumePath() string
+ 	GetVolumeType() string   //TBD
+ }
+
+type BlockUnmounter interface {
+ 	Unmounter
+ 	GetVolumePath() string
+}
+```
+# Mounter binding matrix for statically provisioned volumes:
+
+| PV volumeType | Plugin fstype | PVC volumeType  | Result           |
+| --------------|:-------------:| ---------------:|-----------------:|
+|   unspecified |   unspecified | unspecified     | BIND             |
```

Comment:
@msau42 @screeley44 
I understand. Make sense to me. So updated matrix is this, right?
I changed the orders to avoid cofusion.

|# | PV volumeType | PVC volumeType  |  Result        |
|-:| -------------:| ---------------:| --------------:|
|1| unspecified   |  file            |  BIND     |
|2| unspecified   |  block        |  NO BIND          |
|3| file          |  file           |  BIND          |
|4| file          |  block          |  NO BIND       |
|5| block         |  file           |  NO BIND        |
|6| block         |  block          |  BIND          |